api_app/script_analyzers/observable_analyzers/zoomeye.py

```diff
@@ -0,0 +1,93 @@
+
+        # Setting query filters
+        self.query = {}
+        host_search_filters = [
+            "app",
+            "ver",
+            "device",
+            "os",
+            "service",
+            "cidr",
+            "port",
+            "city",
+            "country",
+            "asn",
+        ]
+        web_search_filters = ["header", "keywords", "desc", "title", "site"]
+        if "query" in additional_config_params:
+            for filter in host_search_filters:
+                self.query[filter] = additional_config_params["query"].get(
+                    filter, "NULL"
+                )
+            for filter in web_search_filters:
+                self.query[filter] = additional_config_params["query"].get(
+                    filter, "NULL"
+                )
+
+        if self.observable_classification == "ip":
+            self.query["ip"] = self.observable_name
+        elif self.observable_classification == "domain" and self.search_type == "host":
+            self.query["hostname"] = self.observable_name
+
+        # Setting paging
+        self.pages = additional_config_params.get("page", "NULL")
+
+        # Setting facets
+        self.facets = additional_config_params.get("facets", "NULL")
+
+        # Setting history
+        self.history = additional_config_params.get("history", "NULL")
+
+        self.__api_key = secrets.get_secret(self.api_key_name)
+
+    def run(self):
+        if not self.__api_key:
+            raise AnalyzerRunException(
+                f"no API key retrieved with name: '{self.api_key_name}'"
+            )
+        if self.search_type == "host" or self.search_type == "web":
+            self.url = self.base_url + self.search_type + "/search?query="
+            queries = []
+            query_filters = self.query.keys()
+            for query_filter in query_filters:
+                if self.query[query_filter] != "NULL":
+                    queries.append(f"{query_filter}:{self.query[query_filter]}")
+            self.url += "%20".join(queries)
+
+            if self.pages != "NULL":
+                self.url += f"&page={self.pages}"
+
+            if self.facets != "NULL":
+                self.url += f"&facet={','.join(self.facets)}"
+
+        elif self.search_type == "both":
+            self.url = self.base_url + "both/search?"
+            if self.history != "NULL":
+                self.url += f"history={self.history}&"
+            self.url += f"ip={self.query['ip']}"
+
+        else:
+            raise AnalyzerConfigurationException(
+                f"search type: '{self.search_type}' not suported."
+                "Supported are: 'host', 'web'."
+            )
+        try:
+            response = requests.get(self.url, headers={"API-KEY": self.__api_key})
+            response.raise_for_status()
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        return response.json()
```

Comment:
please add to the response some keys in the dictionary where is specified which custom options the user provided.

Reply:
You mean the `runtime_config` parameters?